sw/device/mask_rom/boot.md

```diff
@@ -269,3 +314,103 @@ peripheral_lockdown() {
+
+There are two main info structures used by the Mask ROM:
+
+*   The Boot Policy structure, used to choose a ROM_EXT to boot.
+*   The ROM_EXT manifest, used to contain information about a specific ROM_EXT.
+
+In order to keep the Mask ROM simple, a particular Mask ROM version will only
+support one version each of the following structures. This means they must be
+carefully designed to be extensible if the other systems accessing them may
+require additional data in these formats.
+
+### Key Management Structures
+
+Accessed by:
+
+*   Mask ROM (to identify public key is Valid).
+*   ROM_EXT (to prevent use of public key).
+
+Needs to contain:
+
+*   Read-only list of public key ids.
+*   OTP fuses to say whether a key id is revoked or not (keys start not revoked).
+
+### Boot Policy Structure
+
+Accessed by:
+
+*   Mask ROM (to choose ROM_EXT, and during bootstrapping to bless new ROM_EXT).
+*   ROM_EXT (during firmware update).
+
+Needs to contain:
+
+*   Identifier (so we know we're reading the right thing)
+
+    This also acts like a version number, because the Mask ROM code that parses
+    the boot policy can never be updated. Conversely, any changes to this
+    structure require new Mask ROM parsing code, which should be denoted with a
+    new identifier.
+
+*   Which ROM_EXT slot should be chosen first (2.b, 2.c.i).
+*   What to do if ROM_EXT does not validate (2.c.ii, 2.c.iii):
+    *   Try Alternate ROM_EXT; or
+    *   Fail to Boot
+*   What to do if ROM_EXT validates successfully, just before jumping to ROM_EXT:
+    *   Do nothing; or
+    *   Set current ROM_EXT as Primary if not already.
+*   Checksum (of everything else).
+
+Stored in: Flash (Info Partition)
+
+Extensibility: None. This info only controls the actions of the Mask ROM. You
+cannot add functionality to the Mask ROM of a given chip, so there's no way to
+add other information to this structure.
+
+The in-RAM representation also contains the boot reason (reset manager), because
+it needs to be checked most times the boot policy is also read.
+
+### ROM_EXT Manifest Structure
+
+Accessed by:
+
+*   Mask ROM (to validate ROM_EXT)
+*   BL0 Kernel (during firmware update).
+
+Needs to contain:
+
+*   Unsigned Area:
+    *   Identifier (so we know we're reading the right thing)
+    *   Signature
+*   Signed Area:
+    *   Image Length
+    *   Entry Point
+        **Open Q:** Is this fixed or programmable? Field not needed if fixed.
+
+    *   ROM_EXT PMP Region Information
+        **Open Q:** Is this fixed or programmable? Field not needed if fixed.
+
+    *   Key Selection Area
+        *   Public Key
+
+    *   Software Binding Properties:
+        *   ROM_EXT Version
+        *   Usage Constraints
+        *   Peripheral Lockdown Information
+    *   Read-only Extension Area:
+        *   Several words for ROM_EXT/BL0 use only.
+            (Interpretation governed by ROM_EXT/BL0 Version).
+            **Open Q:** Does this satisfy the previous usecases?
```

Comment:
Would it make sense to move this after the code image so that the size is flexible?

Reply:
We actually think this will just be a set of `(pointer, checksum)` pairs, which point to data somewhere inside the image, but store the checksum of that data in the ROM_EXT/BL0 area. This still allows the pointed-to data to be of variable length, as the checksum format/calculation/validation is still up to the ROM_EXT/BL0, but the idea is to strongly suggest the checksum should be used. 

So this will become a fixed number of `(pointer, checksum)` slots, each of which can only be allocated once, ever, in a given ROM_EXT manifest version (each Mask ROM can only interpret one manifest version, but that doesn't mean we shouldn't extend this area as/when needed).